LaunchUpCorp/Focus-Finder
# feat(FOC-12): add api to create user's browsing activity

## feat(FOC-12): add backend API to create user's browsing activity

### Type of Change


### Description

- Created a db table called Activity that stores users' browsing activities.
 - Columns
   - id (auto generated)
   - url
   - startTime (visited timestamp)
   - endTime (left timestamp)
   - userId (foreign key to the user's id)
- Added a GraphQL API that creates a record in the Activity table.

#### Example GraphQL query
```
mutation {
  createActivity(createActivityData: {
    url: "https://google.com",
    startTime: 1683034649000,
    endTime: 1683034750000,
    userId: 3,
  }) {
    id
    url
    startTime
    endTime
    userId
  }
}```

### Testing Instructions

- Make sure PostgreSQL is installed and running on your machine. (See docs for installation: https://www.postgresql.org/download/)
- Create .env file and copy contents in .env.example and update values to your own.
- Run `npm install` to install dependencies.
- Run `npx prisma migrate dev` to apply migrations. (To learn more about Prisma, visit: https://www.prisma.io/docs)
- Run `npm run start`(development) or `npm run start:dev`(watch mode) to run server on `localhost:3000`.
- Visit `localhost:3000/graphql` to test APIs in the playground.
- You can see database records in Prisma Studio by running `npx prisma studio`. For more informaion on Prisma Studio visit https://www.prisma.io/docs/concepts/components/prisma-studio

### Deployment Considerations (optional)

User auth to be implemented later to verify the user for accessing the api.

### Checklist (delete irrelevant items)

- [x] I have followed coding standards and best practices.
- [x] I have updated the relevant tests if needed.
